fieldservice_route_availability/tests/test_route_availability.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2025 APSL-Nagarro Antoni Marroig
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import timedelta
+
+from odoo import fields
+from odoo.exceptions import ValidationError
+from odoo.tests import common
+
+
+class TestRouteAvailability(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.Order = cls.env["fsm.order"]
+        cls.test_person = cls.env.ref("fieldservice.test_person")
+        cls.test_location = cls.env.ref("fieldservice.test_location")
+        cls.blackout_group = cls.env["fsm.blackout.group"].create(
+            {
+                "name": "Test Blackout Group",
+                "fsm_blackout_day_ids": [
+                    (0, 0, {"name": "Test Blackout Day", "date": fields.Date.today()})
+                ],
+            }
+        )
+
+        cls.days = [
+            cls.env.ref("fieldservice_route.fsm_route_day_0").id,
+            cls.env.ref("fieldservice_route.fsm_route_day_1").id,
+            cls.env.ref("fieldservice_route.fsm_route_day_2").id,
+            cls.env.ref("fieldservice_route.fsm_route_day_3").id,
+            cls.env.ref("fieldservice_route.fsm_route_day_4").id,
+            cls.env.ref("fieldservice_route.fsm_route_day_5").id,
+            cls.env.ref("fieldservice_route.fsm_route_day_6").id,
+        ]
+        cls.fsm_route_id = cls.env["fsm.route"].create(
+            {
+                "name": "Demo Route",
+                "max_order": 10,
+                "fsm_person_id": cls.test_person.id,
+                "day_ids": [(6, 0, cls.days)],
+            }
+        )
+        cls.test_location.fsm_route_id = cls.fsm_route_id.id
+        cls.fsm_route_id.fsm_blackout_group_ids = [cls.blackout_group.id]
+
+    def test_blackout_day_validation(self):
+        with self.assertRaises(ValidationError):
```

Comment:
```suggestion
        with self.assertRaisesRegex(ValidationError, r"The date .+ is a blackout day"):
```

Use `self.assertRasesRegex` instead, everywhere.
This lets you be really sure that the exception raised is the one you're looking for, and not a different one raised by another module, for example